### PR TITLE
ATS-237 | Your income tax and national insurance for Scottish persona (rUk income tax table)

### PR DESCRIPTION
**Pull request by Synnnapse**

### CODE REVIEW

File: app/view_models/paye/PayeIncomeTaxAndNics.scala [hunk 1]
```diff
@@ -23,34 +23,46 @@ import view_models.{Amount, Rate}
...
+  lazy val uKRates: List[String] = Play.current.injector.instanceOf[PayeConfig].ukTaxBandKeys
+
   def apply(payeAtsData: PayeAtsData): PayeIncomeTaxAndNics = {
 
-    val scottishTaxBands = (for {
+    PayeIncomeTaxAndNics(payeAtsData.taxYear,
+      getTaxBands(payeAtsData,scottishRates),
+      getTaxBands(payeAtsData,uKRates),
+      getTotalIncomeTax(payeAtsData , "scottish_total_tax"),
+      getTotalIncomeTax(payeAtsData , "total_UK_income_tax"))
```
Comment (oc-ste) [inline app/view_models/paye/PayeIncomeTaxAndNics.scala]: Just want to double check 

> UK

is OK to be capitalised here as everything else is lower case

File: test/config/PayeConfigSpec.scala [hunk 2]
```diff
@@ -65,6 +65,22 @@ class PayeConfigSpec extends UnitSpec with MockitoSugar {
       config.scottishTaxBandKeys shouldBe expected
     }
 
+    "retrieve UK tax band keys in order for a valid year" in {
+      val config = new PayeConfig {
+        override protected val configPath: String = "paye_config.conf"
```
Comment (oc-ste) [inline test/config/PayeConfigSpec.scala]: Why don't you just use paye.conf here? Is paye_config.conf just for testing? If it is, this test doesn't test anything in production code.

File: test/services/atsData/PayeAtsTestData.scala [hunk 3]
```diff
@@ -311,38 +310,45 @@ object PayeAtsTestData {
...
+
+            "basic_rate_income_tax_amount" -> Amount.gbp(380),
+            "basic_rate_income_tax" -> Amount.gbp(2000),
+            "higher_rate_income_tax_amount" -> Amount.gbp(2030),
+            "higher_rate_income_tax" -> Amount.gbp(10150),
+            "ordinary_rate_amount" -> Amount.gbp(4080),
+            "ordinary_rate" -> Amount.gbp(19430),
+            "upper_rate_amount" -> Amount.gbp(12943),
+            "upper_rate" -> Amount.gbp(31570),
+            "total_UK_income_tax" -> Amount.gbp(19433)
```
Comment (oc-ste) [inline test/services/atsData/PayeAtsTestData.scala]: Would it help the tests if Scottish and rUK values were different?

File: app/view_models/paye/PayeIncomeTaxAndNics.scala (hunk 1 above)
Comment (shikkhasingh) [reply app/view_models/paye/PayeIncomeTaxAndNics.scala]: agreed
changed to lower case

File: test/config/PayeConfigSpec.scala (hunk 2 above)
Comment (shikkhasingh) [reply test/config/PayeConfigSpec.scala]: Agreed
deleted the paye_config.conf and now using paye.conf instead for testing

File: test/services/atsData/PayeAtsTestData.scala (hunk 3 above)
Comment (shikkhasingh) [reply test/services/atsData/PayeAtsTestData.scala]: agreed
changed the rUK totalIncome tax values

File: app/view_models/paye/PayeIncomeTaxAndNics.scala [hunk 4]
```diff
@@ -23,34 +23,46 @@ import view_models.{Amount, Rate}
...
+  lazy val uKRates: List[String] = Play.current.injector.instanceOf[PayeConfig].ukTaxBandKeys
+
   def apply(payeAtsData: PayeAtsData): PayeIncomeTaxAndNics = {
 
-    val scottishTaxBands = (for {
+    PayeIncomeTaxAndNics(payeAtsData.taxYear,
+      getTaxBands(payeAtsData,scottishRates),
+      getTaxBands(payeAtsData,uKRates),
+      getTotalIncomeTax(payeAtsData , "scottish_total_tax"),
+      getTotalIncomeTax(payeAtsData , "total_uk_income_tax"))
```
Comment (oc-ste) [inline app/view_models/paye/PayeIncomeTaxAndNics.scala]: @Synnnapse

Comment (Synnnapse) [reply app/view_models/paye/PayeIncomeTaxAndNics.scala]: cheers @oc-ste 
Done 👍

**Pull request merged**